kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriterTest.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    @Test
+    public void testBigQueryNoFailure() {
+        final String topic = "test_topic";
+        final String dataset = "scratch";
+        final Map<String, String> properties = makeProperties("3", "2000", topic, dataset);
+
+        BigQuery bigQuery = mock(BigQuery.class);
+        Map<Long, List<BigQueryError>> emptyMap = mock(Map.class);
+        when(emptyMap.isEmpty()).thenReturn(true);
+
+        InsertAllResponse insertAllResponse = mock(InsertAllResponse.class);
+        when(insertAllResponse.hasErrors()).thenReturn(false);
+        when(insertAllResponse.getInsertErrors()).thenReturn(emptyMap);
+
+        when(bigQuery.insertAll(anyObject()))
+                .thenReturn(insertAllResponse);
+
+        SinkTaskContext sinkTaskContext = mock(SinkTaskContext.class);
+
+        BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, null);
+        testTask.initialize(sinkTaskContext);
+        testTask.start(properties);
+        testTask.put(Collections.singletonList(spoofSinkRecord(topic)));
+        testTask.flush(Collections.emptyMap());
+
+        verify(bigQuery, times(1)).insertAll(anyObject());
+    }
+
+    @Test
+    public void testBigQueryPartialFailure() {
+        final String topic = "test_topic";
+        final String dataset = "scratch";
+        final Map<String, String> properties = makeProperties("3", "2000", topic, dataset);
+
+        Map<Long, List<BigQueryError>> insertErrorMap = mock(Map.class);
+        when(insertErrorMap.isEmpty()).thenReturn(false);
+        when(insertErrorMap.size()).thenReturn(1);
+
+        InsertAllResponse insertAllResponseWithError = mock(InsertAllResponse.class);
+        when(insertAllResponseWithError.hasErrors()).thenReturn(true);
+        when(insertAllResponseWithError.getInsertErrors()).thenReturn(insertErrorMap);
+
+        Map<Long, List<BigQueryError>> emptyMap = mock(Map.class);
+        when(emptyMap.isEmpty()).thenReturn(true);
+
+        InsertAllResponse insertAllResponseNoError = mock(InsertAllResponse.class);
+        when(insertAllResponseNoError.hasErrors()).thenReturn(true);
+        when(insertAllResponseNoError.getInsertErrors()).thenReturn(emptyMap);
+
+        BigQuery bigQuery = mock(BigQuery.class);
+
+        when(bigQuery.insertAll(anyObject()))
+                .thenReturn(insertAllResponseWithError)
+                .thenReturn(insertAllResponseNoError);
+
+        List<SinkRecord> sinkRecordList = new ArrayList<>();
+        sinkRecordList.add(spoofSinkRecord(topic));
+        sinkRecordList.add(spoofSinkRecord(topic));
+
+        SinkTaskContext sinkTaskContext = mock(SinkTaskContext.class);
+
+        BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, null);
+        testTask.initialize(sinkTaskContext);
+        testTask.start(properties);
+        testTask.put(sinkRecordList);
+        testTask.flush(Collections.emptyMap());
+
+        verify(bigQuery, times(2)).insertAll(anyObject());
```

Comment:
I'd also want to verify that the second insertAll was called with just the failed rows.

Reply:
I can do that